client/src/modules/subsidies/subsidies.routes.js

```diff
@@ -1,9 +1,42 @@
 angular.module('bhima.routes')
-  .config(['$stateProvider', function ($stateProvider) {
+  .config(['$stateProvider', $stateProvider => {
     $stateProvider
       .state('subsidies', {
         url         : '/subsidies',
         controller  : 'SubsidyController as SubsidyCtrl',
         templateUrl : 'modules/subsidies/subsidies.html',
+      })
+
+      .state('subsidies.create', {
+        url : '/create',
+        params : {
+          subsidy : { value : {} },
+          creating : { value : true },
+        },
+        onEnter : ['$uibModal', subsidyModal],
+        onExit : ['$uibModalStack', closeModal],
+      })
+
+      .state('subsidies.edit', {
+        url : '/edit',
```

Comment:
I think we typically include the id of the object we are editing in the URL, and then download it in the controller.  This ensures that we can link to it from someplace else in the application.

What do you think of changing this URL to being `/subsidies/:id/edit`, removing the subsidy parameter, and then downloading the subsidy in the modal controller?